### PR TITLE
Adds a transactionCategory to txMeta for use in UI

### LINKED ISSUE #6380: Improve front end methods for identifying transaction type
As demonstrated by the bugs we fixed with 6.3.1, our approach to determining transaction type, and therefore confirmation screen to show, is somewhat fragile.

Improvements were made with 6.3.1, but I wonder if we can go a step further, and identify a transaction type at the time the transaction is created. This could be saved with txData, and then we have an authoritative answer with the transaction at all times.

I believe there will always be 4 types of transactions:
- simple sends
- contract deployments
- contract method calls on contracts for which we know the abi within the metamask codebase
- contract method calls on contracts for which the abi is not known

I believe we can identify all cases without having to rely on a network request. While a network request will be needed to get UI relevant data in the last case (i.e. contract method name), it is not needed to successfully render the right contract on the right confirm screen.

I believe the assigning of transaction type to transaction data can be handle without too much difficulty by the transaction controller.

### PR DESCRIPTION
**Pull request by danjm**
Partially resolves #6380 

This PR modifies transaction controller code so that a new property - `transactionCategory` - is added to `txMeta`. This will be consumed by the UI in an associated PR.

### CODE REVIEW

File: ui/app/components/ui/error-message/error-message.component.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import PropTypes from 'prop-types'
 const ErrorMessage = (props, context) => {
   const { errorMessage, errorKey } = props
   const error = errorKey ? context.t(errorKey) : errorMessage
-
+  console.log('!!!!! props', props)
```
Comment (frankiebee) [inline ui/app/components/ui/error-message/error-message.component.js]: no console.log in prod please use log

File: ui/app/pages/confirm-transaction-base/confirm-transaction-base.component.js [hunk 2]
```diff
@@ -539,6 +539,9 @@ export default class ConfirmTransactionBase extends Component {
     const { name } = methodData
     const { valid, errorKey } = this.getErrorKey()
     const { totalTx, positionOfCurrentTx, nextTxId, prevTxId, showNavigation, firstTx, lastTx, ofText, requestsWaitingText } = this.getNavigateTxData()
+    console.log('!!!!!errorMessage', errorMessage)
+    console.log('!!!!!propsErrorKey', propsErrorKey)
+    console.log('!!!!!errorKey', errorKey)
```
Comment (frankiebee) [inline ui/app/pages/confirm-transaction-base/confirm-transaction-base.component.js]: please use log over console.log

**Review by frankiebee — CHANGES_REQUESTED** [review]
please remove console logs

Would it be possible to append to type instead of adding transactionCategory so something more like 
type: `'${type}#${transactionCategory}'` ? (non blocker just curious)

**Comment by danjm** [comment]
@frankiebee console logs removed

**Comment by chikeichan** [comment]
Code LGTM - the only suggest i have is to update the in-file JS DOC to include the new `code` params.

**Comment by danjm** [comment]
> Would it be possible to append to type instead of adding transactionCategory so something more like
type: '${type}#${transactionCategory}' ? 

@frankiebee yes it is possible? why do you prefer that?

**Review by frankiebee — APPROVED** [review]

**Pull request merged**